### DISCUSSION TITLE
Is there any other car company available other than the Volvo company?

### DISCUSSION THREAD

**Discussion by yukudaa** [langroodi/Adaptive-AUTOSAR#28] [Q&A]
Hi! I have a question.
Is there any other car company available other than the Volvo company?
Can you tell me how to do it if possible?

**Comment by langroodi** [comment]
> Is there any other car company available other than the Volvo company?

Yes, there are; like Mercedes Benz, Telsa, Nissan, Audi, and still more. However, I've never tried any of those.

> Can you tell me how to do it if possible?

Definitely I cannot tell how to do this, because it depends on the API you take and your expectation from the API server. If you want to integrate a new vehicle API to the project, then you can start from the `ExtendedVehicle` class and taking a look into the `tryConfigureRestCommunication` method. That may give you some ideas about how to do the authentication and how to call your new API.

**Reply by yukudaa** [reply]
thank you so much!
